jsonschema/src/keywords/pattern.rs

```diff
@@ -43,21 +41,25 @@ impl Validate for PatternValidator {
         instance_path: &InstancePath,
     ) -> ErrorIterator<'a> {
         if let Value::String(item) = instance {
-            if !self.pattern.is_match(item) {
-                return error(ValidationError::pattern(
-                    instance_path.into(),
-                    instance,
-                    self.original.clone(),
-                ));
+            if let Ok(is_match) = self.pattern.is_match(item) {
+                if !is_match {
+                    return error(ValidationError::pattern(
+                        instance_path.into(),
+                        instance,
+                        self.original.clone(),
+                    ));
+                }
             }
         }
         no_error()
     }
 
     fn is_valid(&self, _: &JSONSchema, instance: &Value) -> bool {
         if let Value::String(item) = instance {
-            if !self.pattern.is_match(item) {
-                return false;
+            if let Ok(is_match) = self.pattern.is_match(item) {
+                if !is_match {
+                    return false;
+                }
```

Comment:
I think that it could be 

```suggestion
                return is_match
```

Reply:
Yes, thanks!